x-pack/plugins/security_solution/server/lib/detection_engine/rule_management/logic/crud/rules_management_client.ts

```diff
@@ -0,0 +1,379 @@
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { RulesClient } from '@kbn/alerting-plugin/server';
+
+import type {
+  RuleCreateProps,
+  RuleObjectId,
+  RuleToImport,
+  PatchRuleRequestBody,
+  RuleUpdateProps,
+} from '../../../../../../common/api/detection_engine';
+
+import type { PrebuiltRuleAsset } from '../../../prebuilt_rules';
+
+import { readRules } from './read_rules';
+import { PrepackagedRulesError } from '../../../prebuilt_rules/api/install_prebuilt_rules_and_timelines/install_prebuilt_rules_and_timelines_route';
+
+import {
+  convertPatchAPIToInternalSchema,
+  convertUpdateAPIToInternalSchema,
+  convertCreateAPIToInternalSchema,
+} from '../../normalization/rule_converters';
+import { transformAlertToRuleAction } from '../../../../../../common/detection_engine/transform_actions';
+import type { RuleAlertType, RuleParams } from '../../../rule_schema';
+import { createBulkErrorObject } from '../../../routes/utils';
+
+export interface CreateRuleOptions {
+  /* Optionally pass an ID to use for the rule document. If not provided, an ID will be generated. */
+  /* This is the ES document ID, NOT the rule_id */
+  id?: string;
+  immutable?: boolean;
+  defaultEnabled?: boolean;
+  allowMissingConnectorSecrets?: boolean;
+}
+
+export interface _UpdateRuleProps {
+  existingRule: RuleAlertType;
+  ruleUpdate: RuleUpdateProps;
+}
+
+export interface _PatchRuleProps {
+  existingRule: RuleAlertType;
+  nextParams: PatchRuleRequestBody;
+}
+
+interface CreateCustomRuleProps {
+  params: RuleCreateProps;
+}
+
+interface CreatePrebuiltRuleProps {
+  ruleAsset: PrebuiltRuleAsset;
+}
+
+type UpdateRuleProps = _UpdateRuleProps;
+
+type PatchRuleProps = _PatchRuleProps;
+
+interface DeleteRuleProps {
+  ruleId: RuleObjectId;
+}
+
+interface UpgradePrebuiltRuleProps {
+  ruleAsset: PrebuiltRuleAsset;
+}
+
+interface ImportRuleOptions {
+  allowMissingConnectorSecrets?: boolean;
+}
+
+interface ImportRuleProps {
+  ruleToImport: RuleToImport;
+  overwriteRules?: boolean;
+  options: ImportRuleOptions;
+}
+
+interface ImportNewRuleProps {
+  ruleToImport: RuleToImport;
+  options: ImportRuleOptions;
+}
+
+interface ImportExistingRuleProps {
+  ruleToImport: RuleToImport;
+  existingRule: RuleAlertType;
+}
+
+export interface IRulesManagementClient {
+  createCustomRule: (createCustomRulePayload: CreateCustomRuleProps) => Promise<RuleAlertType>;
+  createPrebuiltRule: (
+    createPrebuiltRulePayload: CreatePrebuiltRuleProps
+  ) => Promise<RuleAlertType>;
+  updateRule: (updateRulePayload: UpdateRuleProps) => Promise<RuleAlertType>;
+  patchRule: (patchRulePayload: PatchRuleProps) => Promise<RuleAlertType>;
+  deleteRule: (deleteRulePayload: DeleteRuleProps) => Promise<void>;
+  upgradePrebuiltRule: (
+    upgradePrebuiltRulePayload: UpgradePrebuiltRuleProps
+  ) => Promise<RuleAlertType>;
+  importRule: (importRulePayload: ImportRuleProps) => Promise<RuleAlertType>;
```

Comment:
I thought we said during our last discussion that enable and disable didn't make much sense in this interface because they would be just an additional abstraction layer on top of something like `rulesClient.enable(...)` without providing much benefit, like hiding away config options, etc.

Are you thinking about some use cases where it makes sense to add them here?